src/main/java/mate/academy/spring/validation/EmailValidator.java

```diff
@@ -0,0 +1,21 @@
+package mate.academy.spring.validation;
+
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+import mate.academy.spring.validation.annotation.Email;
+
+public class EmailValidator implements ConstraintValidator<Email, String> {
+    private static final String EMAIL_PATTERN = "^(?=.{1,64}@)[A-Za-z0-9_-]"
+            + "+(\\.[A-Za-z0-9_-]+)*@[^-][A-Za-z0-9-]"
+            + "+(\\.[A-Za-z0-9-]+)*(\\.[A-Za-z]{2,})$";
+
+    @Override
+    public boolean isValid(String email,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        if (email == null) {
+            return false;
+        }
+        return Pattern.compile(EMAIL_PATTERN).matcher(email).matches();
```

Comment:
```suggestion
    public boolean isValid(String email,
                           ConstraintValidatorContext constraintValidatorContext) {
        return email != null && email.matches(EMAIL_PATTERN);
```
It would make sense to use Pattern.compile if you want to create constant pattern and compile regex once, but otherwise you can just use string.equals(regex)